scipy/stats/tests/test_stats.py

```diff
@@ -3168,25 +3169,27 @@ def test_zscore_masked_element_0_gh19039(self, xp):
             res = stats.zscore(y, axis=None)
         assert_equal(res[1:], np.nan)
 
-    @pytest.mark.filterwarnings("ignore:divide by zero encountered:RuntimeWarning:dask")
-    @pytest.mark.filterwarnings("ignore:invalid value encountered:RuntimeWarning:dask")
-    def test_degenerate_input(self, xp):
-        scores = xp.arange(3)
-        compare = xp.ones(3)
-        ref = xp.asarray([-xp.inf, xp.nan, xp.inf])
-        warn_ctx = (
-            contextlib.nullcontext() if is_lazy_array(scores)
-            else pytest.warns(RuntimeWarning, match="Precision loss occurred..."))
 
-        with warn_ctx:
-            res = stats.zmap(scores, compare)
-        xp_assert_equal(res, ref)
+@make_skip_xp_backends(stats.gzscore)
+class TestGZscore:
+    def test_gzscore_normal_array(self, xp):
+        x = np.asarray([1, 2, 3, 4])
+        z = stats.gzscore(xp.asarray(x))
+        desired = np.log(x / stats.gmean(x)) / np.log(stats.gstd(x, ddof=0))
+        xp_assert_close(z, xp.asarray(desired, dtype=xp.asarray(1.).dtype))
 
-    @pytest.mark.filterwarnings("ignore:divide by zero encountered:RuntimeWarning:dask")
-    def test_complex_gh22404(self, xp):
-        res = stats.zmap(xp.asarray([1, 2, 3, 4]), xp.asarray([1, 1j, -1, -1j]))
-        ref = xp.asarray([1.+0.j, 2.+0.j, 3.+0.j, 4.+0.j])
-        xp_assert_close(res, ref)
+    @skip_xp_invalid_arg
```

Comment:
For future reference, what does this do now that `xp` is provided automatically? I thought it was enough to leave `xp` out of the argument list.